pkg/matcher/cel.go

```diff
@@ -17,6 +18,24 @@ func celEvaluate(ctx context.Context, expr string, event *info.Event, vcx provid
 	eventTitle := event.PullRequestTitle
 	if event.TriggerTarget == "push" {
 		eventTitle = event.SHATitle
+		// For push event the target_branch & source_branch info coming from payload have refs/heads/
+		// but user may or mayn't provide refs/heads/ info while giving target_branch or source_branch in CEL expression
+		// ex:  pipelinesascode.tekton.dev/on-cel-expression: |
+		//        event == "push" && target_branch == "main" && "frontend/***".pathChanged()
+		// This logic will handle such case.
+		splittedValue := strings.Split(expr, "&&")
+		for i := range splittedValue {
+			if strings.Contains(splittedValue[i], "target_branch") {
+				if !strings.Contains(splittedValue[i], "refs/heads/") {
```

Comment:
should not be hasprefix instead of contains, but i guess that cannot happen to have a branch with refs heads at the end

Reply:
I think we should handle both as well, where the user provide refs/heads and doesn't, since it's documented as such, or we need to adjust the documentation

Reply:
Yes this handles both

If you see

```
if strings.Contains(splittedValue[i], "target_branch") {
				if !strings.Contains(splittedValue[i], "refs/heads/") {
					event.BaseBranch = strings.TrimPrefix(event.BaseBranch, "refs/heads/")
				}
			}
			if strings.Contains(splittedValue[i], "source_branch") {
				if !strings.Contains(splittedValue[i], "refs/heads/") {
					event.HeadBranch = strings.TrimPrefix(event.HeadBranch, "refs/heads/")
				}
			}
```
I m not doing anything if user itself give `refs/heads/` in annotation just using as it is

If user don't give `refs/heads/` in CEL annotation i m trimming that

So basically it handles for with and without `refs/heads `